tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java

```diff
@@ -115,8 +115,7 @@ public boolean hasNext() {
       if (!loadCache()) {
         endOfRegion = true;
```

Comment:
I wonder if we should return false here. Since endOfRegion is true, hasNext should be false.